spec/models/user_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { User.new(name: 'Ali') }
+
+  before { subject.save }
+
+  describe 'validation tests' do
+    it 'name should be present' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should be integer' do
+      subject.posts_counter = 'hey'
+      expect(subject).to_not be_valid
+    end
+
+    it 'posts_counter should be greater than or equal to zero' do
+      subject.posts_counter = -2
+      expect(subject).to_not be_valid
+      subject.posts_counter = 0
+      expect(subject).to be_valid
+    end
+  end
```

Comment:
- [ ] Same thing here. Kindly look into them and fix.

### Test Result

![user-1](https://github.com/adarachel/rails-blog/assets/26597432/ba265c85-d5e4-4b1d-ba9d-ae957afc7649)


![user-2](https://github.com/adarachel/rails-blog/assets/26597432/7f2fafd8-15a5-4898-ab8b-be35257dc59e)